src/decoder/mod.rs

```diff
@@ -1223,116 +1061,46 @@ impl<R: Read + Seek> Decoder<R> {
 
     pub fn read_strip_to_buffer(&mut self, mut buffer: DecodingBuffer) -> TiffResult<()> {
         self.check_chunk_type(ChunkType::Strip)?;
-        let index = self.current_chunk;
-        let offset = *self
-            .image()
-            .chunk_offsets
-            .get(index)
-            .ok_or(TiffError::FormatError(
-                TiffFormatError::InconsistentSizesEncountered,
-            ))?;
-        let byte_count = *self
-            .image()
-            .chunk_bytes
-            .get(index)
-            .ok_or(TiffError::FormatError(
-                TiffFormatError::InconsistentSizesEncountered,
-            ))?;
-        let rows_per_strip =
-            usize::try_from(self.image().strip_decoder.as_ref().unwrap().rows_per_strip)?;
 
-        let sized_width = usize::try_from(self.image().width)?;
-        let sized_height = usize::try_from(self.image().height)?;
-
-        let strip_height_without_padding = index
-            .checked_mul(rows_per_strip)
-            .and_then(|x| sized_height.checked_sub(x))
-            .ok_or(TiffError::IntSizeError)?;
-
-        // Ignore potential vertical padding on the bottommost strip
-        let strip_height = rows_per_strip.min(strip_height_without_padding);
-
-        let buffer_size = sized_width
-            .checked_mul(strip_height)
-            .and_then(|x| x.checked_mul(self.image().bits_per_sample.len()))
-            .ok_or(TiffError::LimitsExceeded)?;
+        let offset = self.image.chunk_file_range(self.current_chunk)?.0;
+        self.goto_offset_u64(offset)?;
 
-        if buffer.len() < buffer_size {
-            return Err(TiffError::FormatError(
-                TiffFormatError::InconsistentSizesEncountered,
-            ));
-        }
+        let byte_order = self.reader.byte_order;
+        let output_width = usize::try_from(self.image().width)?;
+        self.image.expand_chunk(
+            &mut self.reader,
+            buffer.copy(),
+            output_width,
+            byte_order,
+            self.current_chunk,
+        )?;
 
-        self.expand_strip(buffer.subrange(0..buffer_size), offset, byte_count)?;
         self.current_chunk += 1;
 
-        match self.image().predictor {
-            Predictor::None => {}
-            Predictor::Horizontal => {
-                rev_hpredict(
-                    buffer.copy(),
-                    (self.image().width, u32::try_from(strip_height)?),
-                    usize::try_from(self.image().width)?,
-                    self.colortype()?,
-                )?;
-            }
-            Predictor::__NonExhaustive => unreachable!(),
-        }
         Ok(())
     }
 
     fn read_tile_to_buffer(
```

Comment:
Could you make this just be `read_chunk_to_buffer` and have `read_strip_to_buffer` call it?